src/libraries/Microsoft.Extensions.Options/tests/SourceGenerationTests/Baselines/NetFX/Validators.g.cs

```diff
@@ -2128,47 +2128,53 @@ public __SourceGen__RangeAttribute(global::System.Type type, string minimum, str
         public bool ConvertValueInInvariantCulture { get; set; }
         public override string FormatErrorMessage(string name) =>
                 string.Format(global::System.Globalization.CultureInfo.CurrentCulture, GetValidationErrorMessage(), name, Minimum, Maximum);
-        private bool NeedToConvertMinMax { get; }
-        private bool Initialized { get; set; }
-        private const string c_minMaxError = "The minimum and maximum values must be set to valid values.";
+        private readonly bool _needToConvertMinMax;
+        private volatile bool _initialized;
+        private const string MinMaxError = "The minimum and maximum values must be set to valid values.";
 
         public override bool IsValid(object? value)
         {
-            if (!Initialized)
+            if (!_initialized)
             {
-                if (Minimum is null || Maximum is null)
-                {
-                    throw new global::System.InvalidOperationException(c_minMaxError);
-                }
-                if (NeedToConvertMinMax)
+                lock (this)
```

Comment:
Is there any chance that the validation attribute could be locked by user code in a way that could contribute to deadlocks?

Reply:
Also, could the initialization logic + locking be lifted to a helper method to reduce the size of the core validation method?

Reply:
> Is there any chance that the validation attribute could be locked by user code in a way that could contribute to deadlocks?

This is very unlikely to happen. Users will not even hold on to the attribute object instance except if they use like the reflection to get it. I cannot think in a scenario someone will need to do so. 

> Also, could the initialization logic + locking be lifted to a helper method to reduce the size of the core validation method?

I want to reduce the churn here as I want to port this fix to 8.0 servicing.

Reply:
Sure, but generally speaking locking on `this` for types that are accessible from user code is something we should avoid under every circumstance. Including a private `lockObj` field should be simple enough.

Reply:
Ok, I will do that. Thanks.